Python/ReviewSession10/EisAutoAcceptance/matching.py

```diff
@@ -0,0 +1,384 @@
+                )
+            )
+        except TimeoutException:
+            pass
+
+    def exit_receipting(self):
+        self.payment.is_taken = True
+        self.payment.save()
+        self.driver.quit()
+        return self.payment.is_taken
+
+
+class RegisterReceipting(Receipting):
+    def __init__(self, payment):
+        super().__init__(payment)
+        self.reg_payments = self.get_reg_payments()
+
+    def get_reg_payments(self):
+        return (
+            models.RegisterPayment.select()
+            .where(
+                models.RegisterPayment.date == self.payment.date,
+                models.RegisterPayment.num == self.payment.num,
+            )
+            .order_by(models.RegisterPayment.payorder_id)
+            .objects()
+        )
+
+    def fill(self, reg):
+        self.uin_elem.clear()
+        self.driver.find_element_by_name("decisionNum").clear()
+        if not reg.fine:
+            if self.charge_uin.is_selected():
+                WebDriverWait(self.driver, 20).until(
+                    ec.invisibility_of_element_located(
+                        (By.XPATH, '//*[@class="blockUI blockOverlay"]')
+                    )
+                )
+                self.charge_uin.click()
+            self.driver.find_element_by_name("decisionNum").send_keys("09-61")
+        elif reg.fine.uin:
+            if not self.charge_uin.is_selected():
+                WebDriverWait(self.driver, 20).until(
+                    ec.invisibility_of_element_located(
+                        (By.XPATH, '//*[@class="blockUI blockOverlay"]')
+                    )
+                )
+                self.charge_uin.click()
+            self.uin_elem.send_keys(reg.fine.uin)
+        elif reg.fine.number:
+            if self.charge_uin.is_selected():
+                WebDriverWait(self.driver, 20).until(
+                    ec.invisibility_of_element_located(
+                        (By.XPATH, '//*[@class="blockUI blockOverlay"]')
+                    )
+                )
+                self.charge_uin.click()
+            self.driver.find_element_by_name("decisionNum").send_keys(reg.fine.number)
+        self.driver.find_element_by_name("search").click()
+        WebDriverWait(self.driver, 120).until(
+            ec.visibility_of_element_located(
+                (By.XPATH, '//*[@wicketpath="decisionTable_body"]/tr/td[2]/div')
+            )
+        )
+        return False
+
+    def save_payment(self):
+        time_sleep = len(self.reg_payments) * 0.5
+        self.driver.find_element_by_name("save").click()
+        try:
+            WebDriverWait(self.driver, time_sleep).until(
+                ec.presence_of_element_located(
+                    (By.NAME, "table:body:rows:1:cells:1:cell:columnEditorDiv:comp")
+                )
+            )
+        except TimeoutException:
+            pass
+
+    def exit_receipting(self):
+        self.payment.save()
+        self.driver.quit()
+
+
+def receipting(payment):
```

Comment:
This method here is very similar with the one below reg_recipting and i think it could be done all in one. It would make your code neater. 

something like
# use the is_reg as a flag to know the type of action
def receipting(payment, is_reg):
      # Then you add your conditions here